merlin-server/sql/merlin/tables/constraint.sql

```diff
@@ -82,3 +82,20 @@ create trigger set_timestamp
 before update on "constraint"
 for each row
 execute function constraint_set_updated_at();
+
+create or replace function constraint_check_constraint_run()
+returns trigger
+security definer
+language plpgsql as $$begin
+  update constraint_run
+  set status = 'constraint-outdated'
+  where constraint_id = new.id
+  and constraint_definition != new.definition
+  and status = 'resolved';
```

Comment:
Should this trigger also have the opposite update statement? As in, if the status _was_ `constraint-outdated`, but the definitions now match again, it is returned to `resolved`.

Reply:
That's a good point, I forgot that we spoke about this in our initial meeting. I guess that's a larger design question, do we like the idea of resolving past constraint runs if the conditions are met?

Reply:
My gut reaction is yes, because the fact that the status is `constraint-outdated` means that that cached result *would be valid* if the current constraint definition matched the one it was run against. Although revisiting the trigger on `simulation_dataset`, it looks like only the currently-resolved constraints get marked as `simulation_outdated`, not all constraints from that dataset run.

Reply:
Actually, do we need to track `simulation_outdated` as a status? If the sim_dataset_id differs, we _should_ get a cache miss when we try to check for if that constraint definition has been run on the current sim_dataset.

Reply:
But if you can run constraints on against an older sim dataset (which, if it's not part of this PR, is a desired next step that we've discussed with @skovati), then `simulation_outdated` doesn't *mean* anything.

Reply:
That's a good point, I can remove it.